lib/Cpdf.php

```diff
@@ -3614,6 +3811,116 @@ function fillStroke()
         $this->addContent("\nb" . ($this->fillRule === "evenodd" ? "*" : ""));
     }
 
+    /**
+     * add signature
+     *
+     * $fieldSigId = $cpdf->addFormField('Sig', 'Signature1', 0, 0, 0, 0, 0);
+     *
+     * $signatureId = $cpdf->addSignature([
+     *   'signcert' => file_get_contents('dompdf.crt'),
+     *   'privkey' => file_get_contents('dompdf.key'),
+     *   'password' => 'password',
+     *   'name' => 'DomPDF DEMO',
+     *   'location' => 'Home',
+     *   'reason' => 'First Form',
+     *   'contactinfo' => 'info'
+     * ]);
+     * $cpdf->setFormFieldValue($fieldSigId, "$signatureId 0 R");
+     * $cpdf->addForm([$fieldSigId], 3);
+     *
+     * @param array $certInfo (signcert, privkey, password, name, location, reason, contactinfo)
+     * @return int
+     */
+    function addSignature($certInfo = [])
+    {
+        $sigId = ++$this->numObj;
+        $this->o_sig($sigId, 'new', $certInfo);
+
+        return $sigId;
+    }
+
+    /**
+     * add field to form
+     * - type
+     *   Btn => Button Field
+     *   Tx =>  Text Field
+     *   Ch =>  Choice Field
+     *   Sig => Signature
+     *
+     * @param string $type
+     * @param string $name
+     * @param integer $pageNum if 0 then currentpage
+     * @param $x0
+     * @param $y0
+     * @param $x1
+     * @param $y1
+     * @param integer $ff Field Flag
+     * @return int
+     */
+    public function addFormField($type, $name, $pageNum, $x0, $y0, $x1, $y1, $ff = 0)
+    {
+        if ($pageNum > 0) {
+            $pagesObjId = $this->objects[$this->catalogId]['info']['pages'];
+            $pageId = $this->objects[$pagesObjId]['info']['pages'][$pageNum - 1];
+        } else {
+            $pageId = $this->currentPage;
+        }
+
+        $fieldId = ++$this->numObj;
+        $this->o_field($fieldId, 'new', [
+          'rect' => [$x0, $y0, $x1, $y1],
+          'F' => 4,
+          'FT' => "/$type",
+          'T' => "($name)",
+          'Ff' => $ff,
+          'pageid' => $pageId
+        ]);
+
+        return $fieldId;
+    }
+
+    /**
+     * set Field value
+     *
+     * @param $formFieldId
+     * @param $value
+     */
+    public function setFormFieldValue($formFieldId, $value)
```

Comment:
The signature id-reference-notation shouldn't be part of this external method (`$cpdf->setFormFieldValue($fieldSigId, "$signatureId 0 R")`).
Maybe adding a special method to add/map the signature to a field would be a good alternative?